### PR TITLE
[fix](cloud) fix be core when using stream load record in cloud mode

### PR DESCRIPTION
**Pull request by sollhui**
when set enable_stream_load_record=true in cloud mode, meet core:
```
F20240711 00:02:05.733184  2316 storage_engine.cpp:123] Check failed: _type == Type::LOCAL (^A vs. ^@) 
*** Check failure stack trace: ***
    @     0x557f1becd646  google::LogMessageFatal::~LogMessageFatal()
    @     0x557f128ebb9a  doris::BaseStorageEngine::to_local()
    @     0x557f13413005  doris::StreamLoadAction::_save_stream_load_record()
    @     0x557f134125c9  doris::StreamLoadAction::handle()
    @     0x557f1c6399d6  (unknown)
    @     0x557f1c638ee2  (unknown)
    @     0x557f1c61e626  bufferevent_run_readcb_
    @     0x557f1c639f3f  (unknown)
    @     0x557f1c62a679  (unknown)
    @     0x557f1c626e7f  event_base_loop
    @     0x557f13436087  std::_Function_handler<>::_M_invoke()
    @     0x557f12da2148  doris::ThreadPool::dispatch_thread()
    @     0x557f12d970d1  doris::Thread::supervise_thread()
    @     0x7ffb1a784ac3  (unknown)
    @     0x7ffb1a815a04  clone
    @              (nil)  (unknown)
```

### CODE REVIEW

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

File: be/src/http/action/stream_load.cpp [hunk 1]
```diff
@@ -705,8 +706,15 @@ Status StreamLoadAction::_data_saved_path(HttpRequest* req, std::string* file_pa
 
 void StreamLoadAction::_save_stream_load_record(std::shared_ptr<StreamLoadContext> ctx,
                                                 const std::string& str) {
-    auto stream_load_recorder =
-            ExecEnv::GetInstance()->storage_engine().to_local().get_stream_load_recorder();
+    std::shared_ptr<StreamLoadRecorder> stream_load_recorder;
+    if (config::is_cloud_mode()) {
+        stream_load_recorder =
+                ExecEnv::GetInstance()->storage_engine().to_cloud().get_stream_load_recorder();
```
Comment (liaoxin01) [inline be/src/http/action/stream_load.cpp:712]: The get_stream_load_recorder has been implemented in the base class, to_cloud or to_local is not necessary？

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Review by liaoxin01 — DISMISSED** [review]
LGTM

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by sollhui** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by kaijchen — APPROVED** [review]

**Pull request merged**